pkg/controllers/vdb/podfacts.go

```diff
@@ -1184,3 +1184,32 @@ func checkIfNodeUpCmd(podIP string) string {
 	curlCmd := "curl -k -s -o /dev/null -w '%{http_code}'"
 	return fmt.Sprintf("%s %s", curlCmd, url)
 }
+
+// FindFirstPrimaryUpPodIP returns the ip of first pod that
+// has a primary up Vertica node, and a boolean that indicates
+// if we found such a pod
+func (p *PodFacts) FindFirstPrimaryUpPodIP() (string, bool) {
+	initiator, ok := p.findFirstPodSorted(func(v *PodFact) bool {
+		return v.sandbox == vapi.MainCluster && v.isPrimary && v.upNode
+	})
+	return initiator.podIP, ok
```

Comment:
But if initiator is nil, how are we able to follow the pointer to get `podIP`? It would be no different than this, which segfaulted in Go playground:
```
var initiator *PodFact = nil
fmt.Printf("%s\n", initiator.podIP)
```